### PR TITLE
Fix conditional with array parameter

### PR DESCRIPTION
**Pull request by guipublic**
# Description

Array function parameters where not properly conditioned inside an IF branch because the instructions creating the arrays are added during inlining, after the function is compiled.

## Summary of changes

When the function is compiled, we indicate that array parameters are created inside the function entry block.

### CODE REVIEW

File: crates/nargo/tests/test_data/9_conditional/src/main.nr [hunk 1]
```diff
@@ -74,6 +74,10 @@ fn main(a: u32, mut c: [u32; 4], x: [u8; 5], result: pub [u8; 32]){
     constrain c[1] != 2;
     call_intrinsic(x, result);
 
+    //Test case for conditional with arrays from function parameters
+    let b = sort([1,2,3,4]);
+    constrain b[0] == 1;
```
Comment (jfecher) [inline crates/nargo/tests/test_data/9_conditional/src/main.nr:79]: Perhaps we should test this sort function on an unsorted array

**Review by jfecher — APPROVED** [review]

File: crates/nargo/tests/test_data/9_conditional/src/main.nr (hunk 1 above)
Comment (guipublic) [reply crates/nargo/tests/test_data/9_conditional/src/main.nr:79]: This test case is failing without this PR with the array already sorted. This is because it believes it is created in the THEN branch, so it will re-order the array without predicate (i.e not taking into account the condition), which is the correct thing to do (if the array was created in the branch). As a result, this array will not be sorted and the test will fail.
If we use an unsorted array, we will probably get a sorted array and the test will not fail.
Long story short; we need a sorted array here.
